app/lib/analyzer/pana_runner.dart

```diff
@@ -31,27 +31,36 @@ class PanaRunner implements TaskRunner {
 
   @override
   Future<bool> runTask(Task task) async {
-    Summary summary;
-    Directory tempDir;
     final DateTime timestamp = new DateTime.now().toUtc();
-    try {
-      tempDir = await Directory.systemTemp.createTemp('pana');
-      final pubCacheDir = await tempDir.resolveSymbolicLinks();
-      final PackageAnalyzer analyzer = new PackageAnalyzer(
-        flutterDir: envConfig.flutterSdkDir,
-        pubCacheDir: pubCacheDir,
-      );
-      summary = await analyzer.inspectPackage(
-        task.package,
-        version: task.version,
-        keepTransitiveLibs: false,
-      );
-    } catch (e, st) {
-      _logger.severe('Pana execution failed.', e, st);
-    } finally {
-      if (tempDir != null) {
-        await tempDir.delete(recursive: true);
+    Future<Summary> analyze() async {
+      Directory tempDir;
+      try {
+        tempDir = await Directory.systemTemp.createTemp('pana');
+        final pubCacheDir = await tempDir.resolveSymbolicLinks();
+        final PackageAnalyzer analyzer = new PackageAnalyzer(
+          flutterDir: envConfig.flutterSdkDir,
+          pubCacheDir: pubCacheDir,
+        );
+        return await analyzer.inspectPackage(
+          task.package,
+          version: task.version,
+          keepTransitiveLibs: false,
+        );
+      } catch (e, st) {
+        _logger.severe('Pana execution failed.', e, st);
+      } finally {
+        if (tempDir != null) {
+          await tempDir.delete(recursive: true);
+        }
       }
+      return null;
+    }
+
+    Summary summary = await analyze();
+    if (summary == null || (summary.toolProblems?.isNotEmpty ?? false)) {
+      _logger.info('Retrying $task...');
```

Comment:
Ideally I would think this is the exceptional case and we emit it as a warning instead of an info. How often do you think, does this happen?

Reply:
No idea about the frequency. There are two kind of packages that will fall into this: the really outdated ones, which will have the issues even on the retry, and the ones that have only a transient issue, like `pub update` failed. I think there is no reason to call warning here, because there is a lot of old packages that would trigger it.